src/models/CompanyNotification/CompanyNewJobApplicationNotification/Model.ts

```diff
@@ -0,0 +1,56 @@
+import { isDefined, isUUID } from "class-validator";
+import { AttributeNotDefinedError, InvalidAttributeFormatError } from "$models/Errors";
+import { UuidGenerator } from "$models/UuidGenerator";
+
+export class CompanyNewJobApplicationNotification {
+  public uuid: string;
+  public moderatorUuid: string;
+  public companyUuid: string;
+  public jobApplicationUuid: string;
+  public isNew: boolean;
+  public createdAt: Date;
+
+  constructor(attributes: ICompanyNewJobApplicationNotificationAttributes) {
+    this.uuid = attributes.uuid || UuidGenerator.generate();
+    this.moderatorUuid = attributes.moderatorUuid;
+    this.companyUuid = attributes.companyUuid;
+    this.jobApplicationUuid = attributes.jobApplicationUuid;
+    this.isNew = attributes.isNew;
+    this.createdAt = attributes.createdAt;
+    this.validate();
+  }
+
+  private validate() {
+    this.validatePresenceOf();
```

Comment:
lo cambiaria por validate presence, o validate presence of attributes. esta oracion es incompleta